docs/android/emulator/mac-with-windows-vm.md

```diff
@@ -0,0 +1,140 @@
+1. Using `ssh` port forwarding. For more information, see [Use ssh port forwarding](#use-ssh-port-forwarding).
+
+In both cases, the Android Debug Bridge (ADB) is used to connect to the emulator. ADB is a command-line tool that's bundled with the Android SDK Platform Tools package, that lets you communicate with a device. The `adb` command facilitates a variety of device actions, including connecting to devices. For more information about `adb`, see [Android Debug Bridge (adb)](https://developer.android.com/tools/adb) on developer.android.com.
+
+> [!NOTE]
+> You must use an Android Emulator that doesn't include the Google Play Store.
+
+## Use nc or pfctl to perform packet forwarding
+
+To connect to the Android Emulator running on a Mac from a Windows VM, by using packet forwarding, use the following steps:
+
+1. On your Mac, start the Android emulator.
+1. On your Mac, open **Terminal**.
+1. In **Terminal**, kill the `adb` server:
+
+    ```zsh
+    adb kill-server
+    ```
+
+1. In **Terminal**, identify the ports used by the emulator:
+
+    ```zsh
+    lsof -iTCP -sTCP:LISTEN -P | grep 'emulator\|qemu'
+
+    emulator6 94105 macuser   20u  IPv4 0xa8dacfb1d4a1b51f      0t0  TCP localhost:5555 (LISTEN)
+    emulator6 94105 macuser   21u  IPv4 0xa8dacfb1d845a51f      0t0  TCP localhost:5554 (LISTEN)
+    ```
+
+    The emulator uses a pair of sequential TCP ports - an even-numbered port for console connections, and an odd-numbered port for `adb` connections. In the output above the emulator listens for `adb` on port 5555 and listens for console connections on port 5554.
+
+1. In **Terminal**, use either `nc` or `pfctl` to perform packet forwarding:
+
+    1. Use `nc` to forward inbound TCP packets received externally on port 5555 (or any other port) to the odd-numbered port on the loopback interface, and to forward the outbound packets back the other way:
+
+        ```zsh
+        cd /tmp
+        mkfifo backpipe
+        nc -kl 5555 0<backpipe | nc 127.0.0.1 5555 > backpipe
+        ```
+
+        In this example, `127.0.0.1 5555` denotes the odd-numbered port on the loopback interface.
+
+        Provided that the `nc` command keeps running in a Terminal window, packets will be forwarded as expected. Once you've finished using the emulator, you can stop `nc` performing packet forwarding by pressing <kbd>CTRL+C</kbd> in the Terminal window.
+
+        > [!NOTE]
+        > Using `nc` is often preferred to `pfctl`, particularly if the firewall is enabled on your Mac.
+
+    1. Use `pfctl` to redirect TCP packets from port 5555 (or any other port) on the [Shared Networking](https://kb.parallels.com/en/4948#section1) interface to
+    the odd-numbered port on the loopback interface:
+
+        ```zsh
+        sed '/rdr-anchor/a rdr pass on vmnet8 inet proto tcp from any to any port 5555 -> 127.0.0.1 port 5555' /etc/pf.conf | sudo pfctl -ef -
```

Comment:
This is copied exactly from the Xamarin doc, but there's a stray `-` at the end. Remove?

Reply:
I couldn't get the `pfctl` command to work so I'd exclude it unless @BretJohnson can verify. The `nc` command is what I use.

Reply:
I can't get `pfctl` to work either. The `sed` syntax here doesn't appear to be valid and I'm not sure exactly what it's trying to do with this step. Let's just remove it.